### PR TITLE
Rename PipelineResourceResult to RunResult

### PR DESCRIPTION
**Pull request by JeromeJu**
# Changes
This commit renames PipelineResourceResult Golang struct to PipelineInternalResult. No functional changes.

A followup PR for #6434 will move the renamed struct to a proper space since resource_types is no longer used after the removal of all PipelineResources. 

part of https://github.com/tektoncd/pipeline/issues/6197
/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pipeline/v1beta1/resource_types.go [hunk 1]
```diff
@@ -23,12 +23,12 @@ import (
...
 )
 
-// PipelineResourceResult is used to write key/value pairs to TaskRun pod termination messages.
+// TaskAndTerminationResult is used to write key/value pairs to TaskRun pod termination messages.
 // The key/value pairs may come from the entrypoint binary, or represent a TaskRunResult.
 // If they represent a TaskRunResult, the key is the name of the result and the value is the
 // JSON-serialized value of the result.
-// TODO(#6197): Rename this struct
-type PipelineResourceResult struct {
+// #6434 Renamed from PipelineResourceResult
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/resource_types.go:30]: no need to link the issue anymore now that this is done. Please add an alias though for backwards compatibility.

File: pkg/apis/pipeline/v1beta1/resource_types.go [hunk 2]
```diff
@@ -23,12 +23,12 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
-// PipelineResourceResult is used to write key/value pairs to TaskRun pod termination messages.
+// TaskAndTerminationResult is used to write key/value pairs to TaskRun pod termination messages.
```
Comment (lbernick) [inline pkg/apis/pipeline/v1beta1/resource_types.go:26]: I'm struggling to think of a good name here: it's the internal representation of TaskRunResults and TektonInternalResults (and is not necessarily written to termination messages, in the case of sidecar logs). Maybe `GenericResult`? idk

Comment (JeromeJu) [reply pkg/apis/pipeline/v1beta1/resource_types.go]: Thanks, renamed to RunResult and aliased for backward compatibility. 
I was not sure if this Struct should also be moved to result_types.go or somewhere else? I am thinking of a struct in an unversioned package and aliased in v1beta1 if that makes sense.

File: pkg/apis/pipeline/v1/openapi_generated.go [hunk 3]
```diff
@@ -30,68 +30,68 @@ import (
 
 func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
 	return map[string]common.OpenAPIDefinition{
-		"github.com/tektoncd/pipeline/pkg/apis/pipeline/pod.AffinityAssistantTemplate":   schema_pkg_apis_pipeline_pod_AffinityAssistantTemplate(ref),
```
Comment (JeromeJu) [inline pkg/apis/pipeline/v1/openapi_generated.go:33]: also I am confused by how `./hack.update-codegen` has behaved on my local env of m1 macOS, I have put the tekton/pipeline repo under $GOPATH but this seems to be very wrong of changing schema_pkg_apis ref to local 🤔

Comment (lbernick) [reply pkg/apis/pipeline/v1/openapi_generated.go:33]: I'm not sure what the "recommended" approach is but I don't set GOPATH, and the scripts should set up a temporary GOPATH for codegen. Try clearing that env var, running codegen again, and if it's still not working for you, you can open an issue.

File: pkg/apis/pipeline/v1beta1/resource_types.go (hunk 2 above)
Comment (lbernick) [reply pkg/apis/pipeline/v1beta1/resource_types.go]: I think that's fine to leave for a follow-up PR

File: pkg/apis/pipeline/v1/openapi_generated.go (hunk 3 above)
Comment (JeromeJu) [reply pkg/apis/pipeline/v1/openapi_generated.go:33]: Thanks, I have tried unsetting the env variable. But it sill swapped all github paths to my local paths and caused the diff issue. I opened issue to track at https://github.com/tektoncd/pipeline/issues/6485

**Pull request merged**